src/bootstrap/bin/rustc.rs

```diff
@@ -139,6 +139,12 @@ fn main() {
     }
 
     if verbose > 1 {
+        let rust_env_vars =
+            env::vars().filter(|(k, _)| k.starts_with("RUST") || k.starts_with("CARGO"));
```

Comment:
FWIW there's no guarantee that we set them prefixed with RUST or CARGO, though we probably should - it may be worth doing some grep through bootstrap and fixing things up in a separate PR (or commit in this one).

Reply:
hmm.

Are you asking for a future PR that tries to correct any cases we can find of a non-prefixed environment variable?

Or are you asking for something that *enforces* a rule that we do not use environment variables without a `RUST` or `CARGO` prefix ? (Apart from cases like `PATH` ... or perhaps `SHELL` ...)

Reply:
There's quite a bit being set without RUST or CARGO prefixes, mostly starting with CFG_.

<details>

```
$ rg '\.env\("' | grep -v RUST | grep -v CARGO
util.rs:            cmd.env("TERM", "xterm").args(&["--color", "always"]);
dist.rs:    cmd.env("CFG_RELEASE_INFO", builder.rust_version())
dist.rs:        .env("CFG_RELEASE_NUM", &builder.version)
dist.rs:        .env("CFG_RELEASE", builder.rust_release())
dist.rs:        .env("CFG_VER_MAJOR", parts.next().unwrap())
dist.rs:        .env("CFG_VER_MINOR", parts.next().unwrap())
dist.rs:        .env("CFG_VER_PATCH", parts.next().unwrap())
dist.rs:        .env("CFG_VER_BUILD", "0") // just needed to build
dist.rs:        .env("CFG_PACKAGE_VERS", builder.rust_package_vers())
dist.rs:        .env("CFG_PACKAGE_NAME", pkgname(builder, "rust"))
dist.rs:        .env("CFG_BUILD", target.triple)
dist.rs:        .env("CFG_CHANNEL", &builder.config.channel);
dist.rs:        cmd.env("CFG_MINGW", "1").env("CFG_ABI", "GNU");
dist.rs:        cmd.env("CFG_MINGW", "0").env("CFG_ABI", "MSVC");
dist.rs:        cmd.env("CFG_PLATFORM", "x64");
dist.rs:        cmd.env("CFG_PLATFORM", "x86");
builder.rs:            .env("CFG_RELEASE_CHANNEL", &self.config.channel)
builder.rs:        cargo.env("REAL_LIBRARY_PATH_VAR", &util::dylib_path_var());
builder.rs:            cargo.env("REAL_LIBRARY_PATH", e);
builder.rs:            cargo.env("UPDATE_EXPECT", "1");
builder.rs:                cargo.env("CC", cl).env("CXX", cl);
builder.rs:        cargo.env("CFG_COMPILER_HOST_TRIPLE", target.triple);
builder.rs:        cargo.env("CFG_RELEASE_CHANNEL", &self.config.channel);
builder.rs:            cargo.env("WINAPI_NO_BUNDLED_LIBRARIES", "1");
test.rs:        cargo.env("CFG_DISABLE_CROSS_TESTS", "1");
test.rs:        cargo.env("PATH", &path_for_cargo(builder, compiler));
test.rs:            cargo.env("MIRI", &miri);
test.rs:            cargo.env("XARGO_CHECK", builder.out.join("bin").join("xargo-check"));
test.rs:            cargo.env("MIRI_SYSROOT", miri_sysroot);
test.rs:            cargo.env("MIRI", miri);
test.rs:        cargo.env("SYSROOT", builder.sysroot(compiler));
test.rs:        cargo.env("HOST_LIBS", host_libs);
test.rs:        cargo.env("TARGET_LIBS", target_libs);
test.rs:        cargo.env("CLIPPY_DRIVER_PATH", clippy);
test.rs:            .env("CFG_RELEASE_CHANNEL", &builder.config.channel)
test.rs:                    cmd.env("PATH", new_path);
test.rs:        rustbook_cmd.env("PATH", new_path).arg("test").arg(path);
test.rs:            .env("BOOTSTRAP_OUTPUT_DIRECTORY", &builder.config.out)
tool.rs:            cargo.env("LIBZ_SYS_STATIC", "1");
tool.rs:    cargo.env("LZMA_API_STATIC", "1");
tool.rs:    cargo.env("CFG_RELEASE", builder.rust_release());
tool.rs:    cargo.env("CFG_RELEASE_CHANNEL", &builder.config.channel);
tool.rs:    cargo.env("CFG_VERSION", builder.rust_version());
tool.rs:    cargo.env("CFG_RELEASE_NUM", &builder.version);
tool.rs:        cargo.env("CFG_COMMIT_HASH", sha);
tool.rs:        cargo.env("CFG_SHORT_COMMIT_HASH", sha_short);
tool.rs:        cargo.env("CFG_COMMIT_DATE", date);
native.rs:    cfg.env("DESTDIR", "");
native.rs:        cfg.env("SCCACHE_PATH", builder.config.ccache.as_ref().unwrap())
native.rs:            .env("SCCACHE_TARGET", target.triple)
native.rs:            .env("SCCACHE_CC", &cc)
native.rs:            .env("SCCACHE_CXX", &cxx);
native.rs:            cfg.env("SCCACHE_EXTRA_ARGS", "-m32");
native.rs:            .env("LLVM_CONFIG_REAL", &llvm_config)
native.rs:            cfg.env("LLVM_CONFIG_SHIM_REPLACE", &builder.config.build.triple)
native.rs:                .env("LLVM_CONFIG_SHIM_REPLACE_WITH", &target.triple)
compile.rs:        cargo.env("MACOSX_DEPLOYMENT_TARGET", target);
compile.rs:        .env("CFG_RELEASE", builder.rust_release())
compile.rs:        .env("CFG_RELEASE_CHANNEL", &builder.config.channel)
compile.rs:        .env("CFG_VERSION", builder.rust_version())
compile.rs:        .env("CFG_PREFIX", builder.config.prefix.clone().unwrap_or_default());
compile.rs:    cargo.env("CFG_LIBDIR_RELATIVE", libdir_relative);
compile.rs:        cargo.env("CFG_VER_DATE", ver_date);
compile.rs:        cargo.env("CFG_VER_HASH", ver_hash);
compile.rs:        cargo.env("CFG_DISABLE_UNSTABLE_FEATURES", "1");
compile.rs:        cargo.env("CFG_DEFAULT_LINKER", s);
compile.rs:        cargo.env("LLVM_CONFIG", &llvm_config);
compile.rs:            cargo.env("CFG_LLVM_ROOT", s);
compile.rs:            cargo.env("LLVM_LINKER_FLAGS", s);
compile.rs:            cargo.env("LLVM_STATIC_STDCPP", file);
compile.rs:            cargo.env("LLVM_LINK_SHARED", "1");
compile.rs:            cargo.env("LLVM_USE_LIBCXX", "1");
compile.rs:            cargo.env("LLVM_NDEBUG", "1");
```

</details>

Reply:
I think enforcing isn't necessary (and would be painful) but checking rustbuild for it, and perhaps adding a tidy lint that e.g. `env("` in src/bootstrap is always followed by a RUST or CARGO prefix might make sense (we'd opt out when we need to set platform specific variables like ld library path, or maybe allow them explicitly in tidy).